### PR TITLE
Integration for collecting Office 365 logs

### PR DESCRIPTION
**Pull request by Nicogp**
|Related issue|
|---|
|#3812|

## Description

New Office 365 module added to documentation.

## Checks
- [x] It compiles without warnings.
- [x] Spelling and grammar. 
- [x] Used impersonal speech. 
- [x] Used uppercase only on nouns. 
- [x] Updated the `redirect.js` script if necessary (check [this guide](https://github.com/wazuh/wazuh-documentation/blob/master/NEW_RELEASE.md)).

## Note to the reviewer

This PR includes changes to the `redirect.js` script that need to be included in all production branches.

### CODE REVIEW

File: source/office365/monitoring-office365-activity.rst [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+    </office365>
+
+Module reference can be found :ref:`here <office365-module>`.
+
+Using the configuration mentioned above, we will see an example of monitoring Office 365 activity.
+
+Generate activity on Office 365
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+For this example, we will start by generating some activity in our Office 365 Organization, in this case let's add a new mailbox to our team. If we do that, we can see that Office 365 will generate a new json event, something like this:
```
Comment (TomasTurina) [inline source/office365/monitoring-office365-activity.rst:119]: I think this example is not related to the subscription configured above.

File: source/office365/index.rst [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+.. meta::
+  :description: Discover how Wazuh can help you to monitor your Office 365 organizations.
+
+.. versionadded:: 4.4.0
+
+This section provides instructions for monitoring the **Office 365** audit log for your organization.
+
+The **audit** log allows Wazuh to monitor:
+
+- User activity in SharePoint Online and OneDrive for Business
```
Comment (TomasTurina) [inline source/office365/index.rst:17]: Use a `.` to finish each item. Also, is this list complete or we could say that these are the most common events?

File: source/user-manual/reference/ossec-conf/office365-module.rst [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
++----------------------------------------+----------------------------------------------+
+| `api_auth\\client_secret_path`_        | Any string                                   |
++----------------------------------------+----------------------------------------------+
+| `api_auth\\client_secret`_             | Any string                                   |
++----------------------------------------+----------------------------------------------+
+
+api_auth\\tenant_id
+^^^^^^^^^^^^^^^^^^^
+
+Tenat id of your application registered in azure.
```
Comment (TomasTurina) [inline source/user-manual/reference/ossec-conf/office365-module.rst:122]: Azure or Office 365?

File: source/user-manual/reference/ossec-conf/office365-module.rst [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
++--------------------+--------------------+
+| **Default value**  | N/A                |
++--------------------+--------------------+
+| **Allowed values** | Any string         |
++--------------------+--------------------+
+
+api_auth\\client_id
+^^^^^^^^^^^^^^^^^^^
+
+Client id of your application registered in azure.
```
Comment (TomasTurina) [inline source/user-manual/reference/ossec-conf/office365-module.rst:133]: Azure or Office 365?

File: source/user-manual/reference/ossec-conf/office365-module.rst [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
++--------------------+--------------------+
+| **Default value**  | N/A                |
++--------------------+--------------------+
+| **Allowed values** | Any string         |
++--------------------+--------------------+
+
+api_auth\\client_secret_path
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Path of the file that contains the client secret value of your application registered in azure. Incompatible with ``client_secret`` option.
```
Comment (TomasTurina) [inline source/user-manual/reference/ossec-conf/office365-module.rst:144]: Azure or Office 365?

File: source/user-manual/reference/ossec-conf/office365-module.rst [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
++--------------------+--------------------+
+| **Default value**  | N/A                |
++--------------------+--------------------+
+| **Allowed values** | Any string         |
++--------------------+--------------------+
+
+api_auth\\client_secret
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Client secret value of your application registered in azure.
```
Comment (TomasTurina) [inline source/user-manual/reference/ossec-conf/office365-module.rst:155]: Azure or Office 365?

File: source/user-manual/reference/ossec-conf/office365-module.rst [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+The event types to include:
+
+- Audit.AzureActiveDirectory: User identity management.
+- Audit.Exchange: Mail and calendaring server.
+- Audit.SharePoint: Web-based collaborative platform.
+- Audit.General: Includes all other workloads not included in the previous content types.
+- DLP.All: Data loss prevention workloads.
+
++--------------------+-------------------------------------------------------------------------------------------+
+| **Default value**  | N/A                                                                                       |
```
Comment (TomasTurina) [inline source/user-manual/reference/ossec-conf/office365-module.rst:192]: Default value is all five categories.

File: source/user-manual/reference/ossec-conf/office365-module.rst [hunk 8]
```diff
@@ -0,0 +1,240 @@
...
+            <client_id>your_client_id</client_id>
+            <client_secret>your_client_secret</client_secret>
+        </api_auth>
+        <subscriptions>
+            <subscription>Audit.AzureActiveDirectory</subscription>
+            <subscription>Audit.General</subscription>
+        </subscriptions>
+    </office365>
+
+Example of multiple organizations
```
Comment (TomasTurina) [inline source/user-manual/reference/ossec-conf/office365-module.rst:217]:
```suggestion
Example of multiple tenants
```

File: source/user-manual/reference/ossec-conf/office365-module.rst [hunk 9]
```diff
@@ -0,0 +1,240 @@
...
+
++--------------------+--------------------+
+| **Default value**  | N/A                |
++--------------------+--------------------+
+| **Allowed values** | Any string         |
++--------------------+--------------------+
+
+.. note::
+
+    This block can be repeated to give the possibility to connect with more than one organization on Office 365.
```
Comment (TomasTurina) [inline source/user-manual/reference/ossec-conf/office365-module.rst:165]:
```suggestion
    This block can be repeated to give the possibility to connect with more than one tenant on Office 365.
```

File: source/office365/monitoring-office365-activity.rst [hunk 10]
```diff
@@ -0,0 +1,211 @@
...
+.. meta::
+  :description: Discover the way that Wazuh provides to monitor your organization's Office 365 activity.
+
+Monitoring Office 365 Activity
+==============================
+
+The **audit log** allows organization admins to quickly review the actions performed by members of your organization. It includes details such as who performed the action, what the action was, and when it was performed.
+This Wazuh module allows you to collect all the logs from Office 365 using its API. The Office 365 Management Activity API aggregates actions and events into tenant-specific content blobs, which are classified by the type and source of the content they contain. 
+
+List available content:
```
Comment (TomasTurina) [inline source/office365/monitoring-office365-activity.rst:14]: Add subtitle or bold text at least.

File: source/office365/monitoring-office365-activity.rst [hunk 11]
```diff
@@ -0,0 +1,211 @@
...
+
+This operation lists the content currently available for retrieval for the specified content type.
+
+.. code-block:: none
+
+    GET https://manage.office.com/api/v1.0/{client_id}/activity/feed/subscriptions/content?contentType={content_type}&startTime={start_time}&endTime={end_time}
+
+Retrieving content:
```
Comment (TomasTurina) [inline source/office365/monitoring-office365-activity.rst:22]: Add subtitle or bold text at least.

File: source/office365/monitoring-office365-activity.rst [hunk 12]
```diff
@@ -0,0 +1,211 @@
...
+.. code-block:: xml
+
+    GET {content_uri}
+
+Office 365 API description can be found in this `link <https://docs.microsoft.com/en-us/office/office-365-management-api/office-365-management-activity-api-reference>`_.
+
+Office 365 API requirements
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+For **Wazuh** to successfully connect to the **Office365 API**, an authentication process is required. To do this, we must provide the tenant_id, client_id and client_secret of the application that we authorize in the organization.
```
Comment (TomasTurina) [inline source/office365/monitoring-office365-activity.rst:35]: Highlight `tenant_id`, `client_id` and `client_secret`.

File: source/user-manual/reference/ossec-conf/office365-module.rst [hunk 13]
```diff
@@ -0,0 +1,256 @@
...
+.. _office365-module:
+
+office365
+=========
+
+.. versionadded:: 4.3.0
+
+.. note::
+
+    This module only works on Windows, Linux and MacOS. It is recommended to have it enabled only in one agent to avoid repeated logs.
```
Comment (davidfmiranda) [inline source/user-manual/reference/ossec-conf/office365-module.rst:12]:
```suggestion
    This module only works on Windows, Linux, and macOS. It is recommended to have it enabled only in one agent to avoid repeated logs.
```

File: source/user-manual/reference/ossec-conf/office365-module.rst [hunk 14]
```diff
@@ -0,0 +1,256 @@
...
++----------------------------------------+----------------------------------------------+
+| `api_auth\\client_secret_path`_        | Any string                                   |
++----------------------------------------+----------------------------------------------+
+| `api_auth\\client_secret`_             | Any string                                   |
++----------------------------------------+----------------------------------------------+
+
+api_auth\\tenant_id
+^^^^^^^^^^^^^^^^^^^
+
+Tenat id of your application registered in Azure.
```
Comment (davidfmiranda) [inline source/user-manual/reference/ossec-conf/office365-module.rst:136]:
```suggestion
Tenant id of your application registered in Azure.
```

File: source/office365/monitoring-office365-activity.rst [hunk 15]
```diff
@@ -0,0 +1,226 @@
...
+.. code-block:: xml
+
+    GET {content_uri}
+
+Office 365 API description can be found in this `link <https://docs.microsoft.com/en-us/office/office-365-management-api/office-365-management-activity-api-reference>`_.
+
+Office 365 API requirements
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+For **Wazuh** to successfully connect to the **Office365 API**, an authentication process is required. To do this, we must provide the ``tenant_id``, ``client_id`` and ``client_secret`` of the application that we authorize in the organization.
```
Comment (davidfmiranda) [inline source/office365/monitoring-office365-activity.rst:35]:
```suggestion
For **Wazuh** to successfully connect to the **Office365 API**, an authentication process is required. To do this, we must provide the ``tenant_id``, ``client_id``, and ``client_secret`` of the application that we authorize in the organization.
```

File: source/office365/monitoring-office365-activity.rst [hunk 16]
```diff
@@ -0,0 +1,226 @@
...
+2. Certificates & secrets
+
+You can generate a password to use during the authentication process. Go to ``Certificates & secrets`` and click on ``New client secret``:
+
+.. thumbnail:: ../images/office365/3-azure-wazuh-app-create-password.png
+    :title: Certificates & secrets
+    :align: center
+    :width: 100%
+
+.. note:: Make sure you write it down because the UI won’t let you copy it afterwards.
```
Comment (davidfmiranda) [inline source/office365/monitoring-office365-activity.rst:71]:
```suggestion
.. note:: Make sure you write it down because the UI won’t let you copy it afterward.
```

File: source/office365/monitoring-office365-activity.rst [hunk 17]
```diff
@@ -0,0 +1,226 @@
...
+
+3. API permissions
+
+The application needs specific API permissions to be able to request the Office 365 activity events. In this case you are looking for permissions related to the ``https://manage.office.com`` resource.
```
Comment (davidfmiranda) [inline source/office365/monitoring-office365-activity.rst:75]:
```suggestion
The application needs specific API permissions to be able to request the Office 365 activity events. In this case, you are looking for permissions related to the ``https://manage.office.com`` resource.
```

File: source/office365/monitoring-office365-activity.rst [hunk 18]
```diff
@@ -0,0 +1,226 @@
...
+    </office365>
+
+Module reference can be found :ref:`here <office365-module>`.
+
+Using the configuration mentioned above, we will see an example of monitoring Office 365 activity.
+
+Generate activity on Office 365
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+For this example, we will start by generating some activity in our Office 365 Organization, in this case let's modify a ``Communication site`` in ``SharePoint``. If we do that, we can see that Office 365 will generate a new json event, something like this:
```
Comment (davidfmiranda) [inline source/office365/monitoring-office365-activity.rst:120]:
```suggestion
For this example, we will start by generating some activity in our Office 365 Organization. In this case, let's modify a ``Communication site`` in ``SharePoint``. If we do that, we can see that Office 365 will generate a new json event, something like this:
```

**Review by davidfmiranda — CHANGES_REQUESTED** [review]
Hello @Nicogp, I've reviewed your PR, and looks good to me. Nevertheless, I've made you some minor grammar suggestions. 

Regards,

David

**Review by davidfmiranda — APPROVED** [review]
LGTM

**Review by TomasTurina — APPROVED** [review]
LGTM!
